### PR TITLE
refactor: Remove util.inherits #70

### LINKED ISSUE #70: Remove util.inherits
Use ES6 `Class` syntax with `extends` instead of [`utils.inherits`](https://nodejs.org/docs/latest-v14.x/api/util.html#util_util_inherits_constructor_superconstructor)

https://github.com/node-oauth/node-oauth2-server/blob/485147fb156310527456e06da41780b188cf2ca1/lib/grant-types/authorization-code-grant-type.js#L47

https://github.com/node-oauth/node-oauth2-server/blob/485147fb156310527456e06da41780b188cf2ca1/lib/grant-types/client-credentials-grant-type.js#L40

https://github.com/node-oauth/node-oauth2-server/blob/485147fb156310527456e06da41780b188cf2ca1/lib/grant-types/password-grant-type.js#L42

https://github.com/node-oauth/node-oauth2-server/blob/485147fb156310527456e06da41780b188cf2ca1/lib/grant-types/refresh-token-grant-type.js#L47

This also occurs in all the errors at `lib/errors`

### PR DESCRIPTION
**Pull request by dsschiramm**
## Summary
Remove util.inherits
Use ES6 Class syntax with extends instead of utils.inherits

## Linked issue(s)
#70

## Added tests?
Yes, added test to validate OAuthError constructor and properties in test/unit/errors/oauth-error_test.js

## Reproduction
Checkout this branch and run the tests.

### CODE REVIEW

File: lib/errors/access-denied-error.js
Comment (Uzlopak) [inline lib/errors/access-denied-error.js:20]: Use the spread operator 

```javascript
properties  = { 
   code: 400, 
   name: "access_denied",
   ...properties 
};
```

Comment (Uzlopak) [reply lib/errors/access-denied-error.js:20]: This comment is for all error Classes.

File: lib/errors/oauth-error.js
Comment (Uzlopak) [inline lib/errors/oauth-error.js:16]: no need for this if clause just use spread operator. if properties is null, it would still generate the correct Object

File: lib/errors/oauth-error.js
Comment (Uzlopak) [inline lib/errors/oauth-error.js:35]: is this actually necessaray? I think there is actually an issue regarding this unnecessary captureStackTrace call in original repo. As we have now a proper class, the stack trace should be already fine.

File: lib/errors/access-denied-error.js
Comment (Uzlopak) [reply lib/errors/access-denied-error.js:20]: Also remove everywhere the `const util = require('util')`

File: lib/errors/oauth-error.js
Comment (Uzlopak) [inline lib/errors/oauth-error.js:14]: first call super()

**Review by Uzlopak — CHANGES_REQUESTED** [review]
If you follow my comments above the tests will all pass

**Comment by Uzlopak** [comment]
Also please run one time "npm run lint:fix" so that all linting errors go away.

**Comment by Uzlopak** [comment]
@dsschiramm 
Why did you close this?

**Comment by dsschiramm** [comment]
> @dsschiramm Why did you close this?

I made a mistake, I'm fixing it, I'll resubmit the pull request later.

**Comment by Uzlopak** [comment]
You could have kept it open or made it a draft. Then everytime you would push into your repo/branch it would automatically show up here.

**Comment by dsschiramm** [comment]
> You could have kept it open or made it a draft. Then everytime you would push into your repo/branch it would automatically show up here.

Got it, thanks I didn't know, I'll open it and make the corrections.

File: lib/errors/oauth-error.js
Comment (dsschiramm) [reply lib/errors/oauth-error.js]: I just found this issue

https://github.com/oauthjs/node-oauth2-server/issues/436

**Comment by dsschiramm** [comment]
@Uzlopak could you review one more time ?

File: lib/errors/oauth-error.js
Comment (dsschiramm) [reply lib/errors/oauth-error.js]: I suppose that when extending the Error class the act of capturing the stack trace is already done by the Error class constructor.

I will remove it, but I'm not entirely sure about it

**Comment by jankapunkt** [comment]
Hey @dsschiramm thanks a lot for your contribution. Can you please edit your PR and fill out the template sections (summary, linked issues etc.) so we as reviewers can easily step in see what's going on.

File: lib/errors/oauth-error.js
Comment (jankapunkt) [reply lib/errors/oauth-error.js]: I think we should add at least one unit test to check, whether the correct stacktrace is added when raising errors. It could be that `super(message, properties);` messes up the prototype chain so please add one test.

File: lib/errors/oauth-error.js
Comment (jankapunkt) [inline lib/errors/oauth-error.js:28]: Why did you place it exactly at this line and not after the for loop?

File: package.json
Comment (jankapunkt) [inline package.json:67]: Is there a reason for this change in package.json?

**Review by jankapunkt — CHANGES_REQUESTED** [review]

File: package.json
Comment (dsschiramm) [reply package.json:67]: No, I made a change to run the tests and my vscode setting that formats on save is disabled so it doesn't conflict with the project's lint, ended up going to commit accidentally, I will adjust

Comment (jankapunkt) [reply package.json:67]: @dsschiramm thank you

File: lib/errors/oauth-error.js
Comment (dsschiramm) [reply lib/errors/oauth-error.js]: Good question, I changed the place again, put it in the first line of the constructor

**Comment by jankapunkt** [comment]
@dsschiramm we are back in the new year, your PR is not forgotten ;-)

**Comment by jankapunkt** [comment]
I will first create the `4.2.0` release, because it contains important fixes. After that we can check on code-improvements as this one.

**Review by jankapunkt — APPROVED** [review]

**Comment by jankapunkt** [comment]
@Uzlopak as far as I can see all changes were addressed and this can be merged, once you approve

**Comment by jankapunkt** [comment]
@Uzlopak friendly ping

**Review by Uzlopak — APPROVED** [review]
pong ;)

LGTM

**Comment by jankapunkt** [comment]
@dsschiramm it's finally merged :tada: thanks a lot for your contribution and patience!

**Pull request merged**
